### PR TITLE
PBS adapter: fix bug with priceFloors sometimes not being set in request

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Bugfix

## Description of change

﻿This addresses https://github.com/prebid/Prebid.js/issues/8307 by picking what's likely to be the minimum floor for an adUnit (instead of just looking at the floor defined by the first request it finds, which may not be there)

### CODE REVIEW

File: modules/prebidServerBidAdapter/index.js [hunk 1]
```diff
@@ -755,21 +756,64 @@ Object.assign(ORTB2.prototype, {
         deepSetValue(imp, 'ext.prebid.storedauctionresponse.id', storedAuctionResponseBid.storedAuctionResponse.toString());
       }
 
-      const getFloorBid = find(firstBidRequest.bids, bid => bid.adUnitCode === adUnit.code && typeof bid.getFloor === 'function');
+      const floor = (() => {
+        // we have to pick a floor for the imp - here we attempt to find the minimimum floor
```
Comment (patmmccann) [inline modules/prebidServerBidAdapter/index.js:760]: spelling on minimum

**Comment by patmmccann** [comment]
@robertrmartinez could you take this one?

**Comment by robertrmartinez** [comment]
Ahhh yes, makes sense this bug existed!

**Review by robertrmartinez — APPROVED** [review]
Looks good, Why the decision to take the min?

I don't really know to be honest what matters just curious.

Could have also just grabbed the first valid floor found I guess.

Or the max?

Does not really matter to me, most of the time it will prob always be the same!

**Comment by dgirardi** [comment]
@robertrmartinez my reasoning is that if you don't pick the minimum one you might lose bids that would have otherwise been available - say that you somehow manage to set a floor of `1` for `bidderA` and `2` for `bidderB`; if the request to PBS sets `bidfloor: 2` and neither bidder can beat it, you won't get any bid. But `bidderA` might be able to beat `1`, which the publisher has indicated as acceptable.

But I'm also not sure this makes sense - maybe the revenue from when they can beat the higher floor outweigh the lost bids.

If noone thinks this is necessary, I'd rather remove the added complexity and just pick the first one.

**Comment by patmmccann** [comment]
i find the reason to use the min here persuasive, seems safer

**Comment by robertrmartinez** [comment]
Sounds good to me!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**